Servers/UI/OJS.Servers.Ui/ClientApp/src/pages/contests/contest-solution-submit/ContestSolutionSubmitPage.tsx

```diff
@@ -338,23 +363,9 @@ const ContestSolutionSubmitPage = () => {
     ]);
 
     useEffect(() => {
-        if (!selectedSubmissionType) {
-            return;
-        }
-
-        const isStrategyFileUpload = selectedSubmissionType?.allowBinaryFilesUpload;
-
-        const isCodeStrategyAndCodeIsEmptyOrTooShort =
-            !isStrategyFileUpload && (isNilOrEmpty(submissionCode) || submissionCode!.length < 5);
-        const isFileUploadAndFileIsEmpty = isStrategyFileUpload && isNil(uploadedFile);
-
-        if (isCodeStrategyAndCodeIsEmptyOrTooShort || isFileUploadAndFileIsEmpty) {
-            setIsSubmitButtonDisabled(true);
-            return;
-        }
-
-        setIsSubmitButtonDisabled(false);
-    }, [ selectedSubmissionType, submissionCode, uploadedFile ]);
+        // Disable submit button when code is updated
+        handleSubmitButtonShouldBeDisabled();
+    }, [ handleSubmitButtonShouldBeDisabled, submissionCode, uploadedFile ]);
```

Comment:
Are `submissionCode` and `uploadFile` needed? If they are, so should be the `selectedSubmissionType`

Reply:
Yes, they should be dependencies as we need to retrigger this function each time either code value changes. I'll add submission type also

Reply:
Fixed